web/src/compose_validate.ts

```diff
@@ -622,30 +654,36 @@ export function validate_stream_message_address_info(sub: StreamSubscription): b
     return false;
 }
 
-function validate_stream_message(scheduling_message: boolean): boolean {
-    const stream_id = compose_state.stream_id();
+function validate_stream_message(scheduling_message: boolean, show_banner = true): boolean {
     const $banner_container = $("#compose_banners");
+    const stream_id = compose_state.stream_id();
     const no_stream_mentioned = stream_id === undefined;
+    set_no_stream_mentioned(no_stream_mentioned);
     if (no_stream_mentioned) {
-        compose_banner.show_error_message(
-            $t({defaultMessage: "Please select a channel."}),
-            compose_banner.CLASSNAMES.missing_stream,
-            $banner_container,
-            $("#compose_select_recipient_widget_wrapper"),
-        );
+        if (show_banner) {
+            compose_banner.show_error_message(
+                $t({defaultMessage: "Please select a channel."}),
+                compose_banner.CLASSNAMES.missing_stream,
+                $banner_container,
+                $("#compose_select_recipient_widget_wrapper"),
+            );
+        }
         return false;
     }
 
     if (realm.realm_mandatory_topics) {
         const topic = compose_state.topic();
         const no_topic_mentioned = topic === "";
+        set_no_topic_mentioned(no_topic_mentioned);
         if (no_topic_mentioned) {
-            compose_banner.show_error_message(
-                $t({defaultMessage: "Topics are required in this organization."}),
-                compose_banner.CLASSNAMES.topic_missing,
-                $banner_container,
-                $("input#stream_message_recipient_topic"),
-            );
+            if (show_banner) {
+                compose_banner.show_error_message(
+                    $t({defaultMessage: "Topics are required in this organization."}),
+                    compose_banner.CLASSNAMES.topic_missing,
+                    $banner_container,
+                    $("input#stream_message_recipient_topic"),
+                );
+            }
             return false;
```

Comment:
This pattern of checking `show_banner` in each of these feels repetitive. I think there should be a better abstraction, where we do `return report_validation_error(..., show_banner)` and don't need so many repeated conditionals.